simphony/cuds/abstractlattice.py

```diff
@@ -22,6 +23,50 @@ class ABCLattice(object):
     """
     __metaclass__ = ABCMeta
 
+    def __init__(self, size, origin, a, b, c, alpha, beta, gamma):
+        """
+        Sets lattice constants a,b,c and angles between primitive vectors
+        """
+        self._size = size
+        self._origin = origin
+        self._a = a
+        self._b = b
+        self._c = c
+        self._alpha = alpha
+        self._beta = beta
+        self._gamma = gamma
+        self._set_primitive_vectors()
+        return self
+
+    def _set_primitive_vectors(self):
+        """
+        Calculates primitive vectors based on the lattice constants and angles. Set the type as
+        TRICLINIC_3D (default).
+        """
+        bcosg = self._b * np.cos(np.radians(self._gamma))
+        bsing = self._b * np.sin(np.radians(self._gamma))
+        ccosb = self._c * np.cos(np.radians(self._beta))
+        ccosa = self._c * np.cos(np.radians(self._alpha))
+        p1 = (self._a, 0, 0)
+        p2 = (bcosg, bsing, 0)
+        p3 = (ccosb, (self._b*ccosa-bcosg*ccosb) / bsing,
+              np.sqrt(np.power(self._c,2) - (np.power(ccosb,2) +
+              np.power((self._b*ccosa-bcosg*ccosb) / bsing,2))))
+        self._primitive_vectors = (p1, p2, p3)
+        self._type = 13#TRICLINIC_3D #default behaviour
+
+    def create_cubic_3d(self, size, origin, a):
```

Comment:
However, for lattice-boltzmann HPC applications, we should be able to construct hdf5-backed lattice without generating the whole lattice in memory first, because often the lattice is much bigger than can be fit into a single PC memory. This doesn't require the use of 'group' parameter, but we should still allow generation of hdf5 backed lattice somehow, because currently it's not supported inside H5CUDS (that only has get.../add... methods for CUDS objects).